File: python/Leetcode1448.py
from Utils import TreeNode
import math

class Leetcode1448:
    def goodNodes(self, root: TreeNode) -> int:
        largest = -math.inf

        def traverse(root, largest):
            if root is None:
                return
            good_nodes = 0
            if root.val >= largest:
                good_nodes += 1
                largest = root.val
                print('Node: ', root.val)            
            return good_nodes + traverse(root.left, largest) + traverse(root.right, largest)

        return traverse(root, largest)

File: java/src/Leetcode1448.java
import utils.TreeNode;

public class Leetcode1448 {
    public int goodNodes(TreeNode root) {
        int largest = Integer.MIN_VALUE;

        return traverse(root, largest);
    }
    private int traverse(TreeNode root, int largest){
        if(root == null)
            return 0;
        int goodNode = 0;
        if(root.val >= largest){
            goodNode++;
            largest = root.val;
        }
        return goodNode + traverse(root.left, largest) + traverse(root.right, largest);
    }
}
